Sources/EpoxyCore/SwiftUI/EpoxySwiftUIHostingView.swift

```diff
@@ -366,7 +368,23 @@ struct EpoxyHostingWrapper<Content: View>: View {
 
   var body: some View {
     content.rootView
-      .environment(\.epoxyLayoutMargins, environment.layoutMargins)
+      .environment(\.epoxyHostingController, environment.hostingController)
       .environment(\.epoxyIntrinsicContentSizeInvalidator, environment.intrinsicContentSizeInvalidator)
+      .environment(\.epoxyLayoutMargins, environment.layoutMargins)
   }
 }
+
+// MARK: - EnvironmentValues
+
+extension EnvironmentValues {
+  public var epoxyHostingController: UIViewController? {
+    get { self[EpoxyHostingControllerKey.self] }
+    set { self[EpoxyHostingControllerKey.self] = newValue }
+  }
+}
+
+// MARK: - EpoxyHostingControllerKey
+
+private struct EpoxyHostingControllerKey: EnvironmentKey {
+  static let defaultValue: UIViewController? = nil
+}
```

Comment:
Could we move these out into another file similar to the EpoxySwiftUIIntrinsicContentSizeInvalidator? Just to make it more discoverable and add some doc comments to the public method